### PR TITLE
Add codeql and checkstyle

### PR DESCRIPTION
**Pull request by cpeeyush**
## What this PR changes/adds

- Run checkstyle on PRs in both upstream and forks.
- Run CodeQL on  PRs in both upstream and forks.

workflow failure due to checkstyle as an example: https://github.com/eclipse-dataspaceconnector/MinimumViableDataspace/runs/7837971765?check_suite_focus=true

## Why it does that

Both Checkstyle and CodeQL are already in use in the EDC framework, so these tools are the natural choice in related projects like the Identity Hub as well.

## Linked Issue(s)

Relates to https://github.com/agera-edc/MinimumViableDataspace/issues/252

## Checklist

- [ ] added appropriate tests?
- [x] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly?

### CODE REVIEW

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -15,22 +15,13 @@ jobs:
...
       - uses: actions/checkout@v2
+      - uses: ./.github/actions/gradle-setup
 
       - name: Check Terraform files are properly formatted (run "terraform fmt -recursive" to fix)
         run: |
           terraform fmt -recursive
           git diff --exit-code
 
-      # run Checkstyle explicitly (as opposed to within gradle) due to better reporting capabilities
       - name: Run Checkstyle
```
Comment (algattik) [inline .github/workflows/check.yaml]: nit: I would suggest running this before `terraform fmt`, since by experience it fails much more often

**Review by algattik — APPROVED** [review]
I tested deriving a PR from this with a checkstyle violation, and that the run indeed fails: https://github.com/eclipse-dataspaceconnector/MinimumViableDataspace/runs/7837971765?check_suite_focus=true

It would have been useful to provide such a sample run as part of the PR.

**Comment by cpeeyush** [comment]
> I tested deriving a PR from this with a checkstyle violation, and that the run indeed fails: https://github.com/eclipse-dataspaceconnector/MinimumViableDataspace/runs/7837971765?check_suite_focus=true
> 
> It would have been useful to provide such a sample run as part of the PR.

Thanks @algattik, yeah good idea, I missed it. Now updated PR description with the failed run as you shared.

File: .github/workflows/check.yaml (hunk 1 above)
Comment (cpeeyush) [reply .github/workflows/check.yaml]: Applied. Thanks

File: docs/developer/decision-records/2022-08-15-code-quality-tooling/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Code Quality
+
+## Decision
+
+Use [Checkstyle](https://checkstyle.sourceforge.io/) and [CodeQL](https://codeql.github.com/) static code analysis tools to guarantee adherence to coding standards and discover potential vulnerabilities.
+
+## Rationale
+
+Both Checkstyle and CodeQL are already in use in the EDC framework, so these tools are the natural choice in related projects like the Identity Hub as well.
```
Comment (marcgs) [inline docs/developer/decision-records/2022-08-15-code-quality-tooling/README.md]:
```suggestion
Both Checkstyle and CodeQL are already in use in the EDC framework, so these tools are the natural choice in related projects like the MVD as well.
```

Comment (ouphi) [inline docs/developer/decision-records/2022-08-15-code-quality-tooling/README.md]:
```suggestion
Both Checkstyle and CodeQL are already in use in the EDC framework, so these tools are the natural choice in related projects like the MinimumViableDataspace as well.
```

**Review by marcgs — APPROVED** [review]

**Review by ouphi — APPROVED** [review]

File: docs/developer/decision-records/2022-08-15-code-quality-tooling/README.md (hunk 2 above)
Comment (cpeeyush) [reply docs/developer/decision-records/2022-08-15-code-quality-tooling/README.md]: Applied similar feedback from Marc. Resolving this. Thanks

**Pull request merged**